@material-ui/x-grid-data-generator not working

<!-- Provide a general summary of the issue in the Title above -->
I've installed the XGrid and the x-grid-data-generator to test it following the example from the documentation, however, after implementing it, it breaks the application completely.

<!-- Checked checkbox should look like this: [x] -->

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
```
ERROR in C:/www/<app>/node_modules/@material-ui/x-grid-data-generator/dist/esm/index.js
Module not found: Error: Can't resolve '@material-ui/lab/Rating' in 'C:\www\<app>\node_modules\@material-ui\x-grid-data-generator\dist\esm'
```

## Expected Behavior 🤔
The application should render and show something like this
![image](https://user-images.githubusercontent.com/77624664/114678344-7189f600-9d0b-11eb-9922-0644fd9402fe.png)


<!-- Describe what should happen. -->

## Steps to Reproduce 🕹

Steps:

1. Install dependencies:
    "@material-ui/data-grid": "4.0.0-alpha.24",
    "@material-ui/icons": "~4.11.2",
    "@material-ui/x-grid": "4.0.0-alpha.24",
    "@material-ui/x-grid-data-generator": "4.0.0-alpha.24",
2. Implement the code from the documentation
3. Observe the error

The code from the documentation
```ts
import * as React from 'react';
import { XGrid } from '@material-ui/x-grid';
import { useDemoData } from '@material-ui/x-grid-data-generator';

export default function XGridDemo() {
  const { data } = useDemoData({
    dataSet: 'Commodity',
    rowLength: 100000,
  });

  return (
    <div style={{ height: 520, width: '100%' }}>
      <XGrid
        {...data}
        loading={data.rows.length === 0}
        rowHeight={38}
        checkboxSelection
      />
    </div>
  );
}
```

## Context 🔦
I am evaluating the XGrid for the project in my company.

## Your Environment 🌎

<!--
  Run `npx @material-ui/envinfo` and post the results.
  If you encounter issues with TypeScript please include the used tsconfig.
-->
<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
System:
    OS: Windows 10 10.0.17763
  Binaries:
    Node: 14.16.0 - ~\AppData\Local\Volta\tools\image\node\14.16.0\node.EXE
    Yarn: 1.22.10 - ~\AppData\Local\Volta\tools\image\yarn\1.22.10\bin\yarn.CMD
    npm: 6.14.11 - ~\AppData\Local\Volta\tools\image\node\14.16.0\npm.CMD
  Browsers:
    Chrome: 89.0.4389.114
    Edge: Not Found
  npmPackages:
    @emotion/styled:  10.0.27
    @material-ui/core: 4.11.3 => 4.11.3
    @material-ui/data-grid: 4.0.0-alpha.24 => 4.0.0-alpha.24
    @material-ui/icons: ~4.11.2 => 4.11.2
    @material-ui/styles:  4.11.3
    @material-ui/system:  4.11.3
    @material-ui/types:  5.1.0
    @material-ui/utils:  4.11.2
    @material-ui/x-grid: 4.0.0-alpha.24 => 4.0.0-alpha.24
    @material-ui/x-grid-data-generator: 4.0.0-alpha.24 => 4.0.0-alpha.24
    @material-ui/x-license:  4.0.0-alpha.21
    @types/react: 17.0.2 => 17.0.2
    react: 17.0.1 => 17.0.1
    react-dom: 17.0.1 => 17.0.1
    styled-components: 5.2.1 => 5.2.1
    typescript: 4.2.3 => 4.2.3
```
Browser: Chrome
</details>

--- a/packages/grid/x-grid-data-generator/package.json
+++ b/packages/grid/x-grid-data-generator/package.json
@@ -29,15 +29,15 @@
     "@types/chance": "^1.1.0",
     "chance": "^1.1.6",
     "clsx": "^1.0.4",
-    "lru-cache": "^6.0.0",
-    "react": "^17.0.0"
+    "lru-cache": "^6.0.0"
   },
   "devDependencies": {
     "esm": "^3.2.25",
     "yargs": "^16.2.0"
   },
   "peerDependencies": {
-    "@material-ui/core": "^4.9.12 || ^5.0.0-alpha.22"
+    "@material-ui/core": "^4.9.12 || ^5.0.0-alpha.22",
+    "react": "^17.0.0"
   },
   "setupFiles": [
     "<rootDir>/src/setupTests.js"

--- a/packages/grid/x-grid-data-generator/package.json
+++ b/packages/grid/x-grid-data-generator/package.json
@@ -24,7 +24,6 @@
     "typescript": "tsc -p tsconfig.json"
   },
   "dependencies": {
-    "@material-ui/core": "^4.9.12 || ^5.0.0-alpha.22",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "^4.0.0-alpha.54",
     "@types/chance": "^1.1.0",
@@ -36,6 +35,9 @@
   "devDependencies": {
     "esm": "^3.2.25",
     "yargs": "^16.2.0"
+  },
+  "peerDependencies": {
+    "@material-ui/core": "^4.9.12 || ^5.0.0-alpha.22"
   },
   "setupFiles": [
     "<rootDir>/src/setupTests.js"

--- a/packages/grid/x-grid-data-generator/package.json
+++ b/packages/grid/x-grid-data-generator/package.json
@@ -24,20 +24,18 @@
     "typescript": "tsc -p tsconfig.json"
   },
   "dependencies": {
+    "@material-ui/core": "^4.9.12 || ^5.0.0-alpha.22",
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.54",
     "@types/chance": "^1.1.0",
     "chance": "^1.1.6",
     "clsx": "^1.0.4",
-    "lru-cache": "^6.0.0"
+    "lru-cache": "^6.0.0",
+    "react": "^17.0.0"
   },
   "devDependencies": {
     "esm": "^3.2.25",
     "yargs": "^16.2.0"
-  },
-  "peerDependencies": {
-    "@material-ui/core": "^4.9.12 || ^5.0.0-alpha.22",
-    "@material-ui/icons": "^4.9.1",
-    "@material-ui/lab": "^4.0.0-alpha.54",
-    "react": "^17.0.0"
   },
   "setupFiles": [
     "<rootDir>/src/setupTests.js"

